OpenFn/kit
# Update pull and deploy to work together

## Short Description

This PR updates pull and deploy to work together.

@josephjclark , I'd like your fresh eyes on the logic and _then_ I'd like your help writing tests and making the types not suck. (I've used that `any` escape hatch which you definitely should never have let me see!)

## Related issue

Fixes #378

## QA Notes

To test this:
1. First `pull` a project from a Lightning server and then run `deploy` on that same project. You should see no changes.
2. Then `deploy` a new project to a Lightning server, committing your `project.yaml` and new `projectState.json` to git. Then run `pull` with the project ID that you've just created. You should see no change in your diff between the yaml or the project state.

## Checklist before requesting a review


## Release branch checklist

If this is a release branch:

- [ ] Ensure versions have been bumped in package.json
- [ ] Run `pnpm install` after bumping versions
- [ ] Ensure tags have been pushed
